server/cache/arc.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"sync"
+)
+
+// ARC is a thread-safe fixed size Adaptive Replacement Cache.
+// ARC is an enhancement over the standard LRU cache in that tracks both
```

Comment:
What's the difference of your implementation and https://github.com/moovweb/go-cache ?